config/routes.rb

```diff
@@ -23,11 +25,15 @@
     # Newer implementation: will supersede the above.
     get "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#show"
     put "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#update"
-    get "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#show"
-    post "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#update"
+
+    get "users/two_factor_authentication/email/resend", to: "users/two_factor_authentication/email#resend"
+    get "users/two_factor_authentication/email/confirm", to: "users/two_factor_authentication/email#confirm"
```

Comment:
IMHO this path should ideally be nested within `/setup` as it's part of the setup process (e.g. `/setup/email?action=confirm`), and would have preferred to have the action (i.e. `confirm`) not specified as an entity on the path.

Reply:
I'm happy with your suggestion to replace `confirm` with `confirmation`. But would it be possible to have it nested within `setup`?

Reply:
I moved both to setup as requested